Riot/Modules/Onboarding/AuthenticationCoordinator.swift

```diff
@@ -611,3 +652,23 @@ extension AuthenticationCoordinator {
         // unused
     }
 }
+
+// MARK: - AuthFallBackViewControllerDelegate
+extension AuthenticationCoordinator: AuthFallBackViewControllerDelegate {
+    @MainActor func authFallBackViewController(_ authFallBackViewController: AuthFallBackViewController,
+                                    didLoginWith loginResponse: MXLoginResponse) {
+        let credentials = MXCredentials(loginResponse: loginResponse, andDefaultCredentials: nil)
+        let client = MXRestClient(credentials: credentials)
+        guard let session = MXSession(matrixRestClient: client) else {
+            MXLog.error("[AuthenticationCoordinator] authFallBackViewController:didLogin: session could not be created")
+            return
+        }
+        let flow: AuthenticationFlow = initialScreen == .login ? .login : .register
+        onSessionCreated(session: session, flow: flow)
```

Comment:
I *think* that marking this delegate methods as `@MainActor` means it doesn't fully conform. Removing that and updating this to be
```suggestion
        Task { await onSessionCreated(session: session, flow: flow) }
```
should do the trick.

Reply:
If its not complaining I think it is because it comes from objc. From what I've read there's basically no support for mapping into structured concurrency from objc.